subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/PathUtil.java

```diff
@@ -219,51 +233,81 @@ public static int compareWithCommonPrefix(String path1, String path2, int offset
                 }
             }
         }
-        if (path1.length() == path2.length() - offset) {
+        if (relativePath.length() == absolutePath.length() - offset) {
             return accumulatedValue;
         }
-        if (path1.length() > path2.length() - offset) {
-            return isFileSeparator(path1.charAt(maxPos)) ? accumulatedValue : 1;
+        if (relativePath.length() > absolutePath.length() - offset) {
+            return isFileSeparator(relativePath.charAt(maxPos)) ? accumulatedValue : 1;
+        }
+        return isFileSeparator(absolutePath.charAt(maxPos + offset)) ? accumulatedValue : -1;
+    }
+
+    /**
+     * Checks whether the relative path given by the absolute path and the offset has the given prefix.
+     */
+    public static boolean isPrefix(String prefix, String absolutePath, int offset, CaseSensitivity caseSensitivity) {
```

Comment:
Call it `hasPrefix`?

Reply:
BTW, this will be so much more understandable on its own type. Compare having to carry all those three variables around and writing this:

```java
PathUtil.hasPrefix(prefix, absolutePath, offset, caseSensitive)
```

to

```java
relativePath.hasPrefix(prefix)
```

I mean, this is what OOP was invented to do. :)

Reply:
Yes, and that is what I have on https://github.com/gradle/gradle/pull/11378.